### PR TITLE
An evil test for slivers.

### PR DESCRIPTION
**Pull request by Hixie**
Combines many sliver features at once, and verifies that it doesn't crash.

### CODE REVIEW

**Comment by Hixie** [comment]
(This caught a lot of errors during development.)

File: packages/flutter/test/widgets/slivers_evil_test.dart [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            new SliverToBoxAdapter(child: new Container(height: 520.0)),
+            new SliverToBoxAdapter(child: new Container(height: 520.0)),
+            new SliverToBoxAdapter(child: new Container(height: 5.0)),
+          ],
+        ),
+      ),
+    );
+    AbsoluteScrollPosition position = scrollableKey.currentState.position;
+    position.animate(to: 10000.0, curve: Curves.linear, duration: const Duration(minutes: 1));
+    await tester.pumpUntilNoTransientCallbacks(const Duration(milliseconds: 10));
```
Comment (abarth) [inline packages/flutter/test/widgets/slivers_evil_test.dart]: Can we change the combination of 1 minute and 10 millisecond intervals?  That seems like it's going to be pretty slow...

**Comment by abarth** [comment]
LGTM

File: packages/flutter/test/widgets/slivers_evil_test.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/test/widgets/slivers_evil_test.dart]: Ah yeah, I meant to tune this down a bit

Comment (Hixie) [inline packages/flutter/test/widgets/slivers_evil_test.dart]: I've made them pump 10,10,50,122...

**Pull request merged**